ui/components/test.jsx

```diff
@@ -1,39 +1,63 @@
-var React = require('react/addons'),
-    Details = require('./details.jsx');
-
-var Test = React.createClass({
-  statics: {
-    DETAILS_TOGGLE_CLASS: 'toggled'
-  },
-  getInitialState: function() {
-    return {toggled: false};
-  },
-  displayDetails: function() {
-    this.setState({toggled: !this.state.toggled});
-  },
-  render: function() {
-    var test = this.props.test,
-        cx = React.addons.classSet,
-        classes = cx({
-          test: true,
-          toggled: this.state.toggled
-        }),
-        paths = {
-          baseline: test.result.baseline
-        };
-
-    if (test.result.screenshot && test.result.diff) {
-      paths.diff = test.result.diff;
-      paths.screenshot = test.result.screenshot;
+import React from 'react';
+import classNames from 'classnames';
+import Details from './details.jsx';
+
+class Test extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {toggled: false};
+
+    this.onDetailsDisplay = this.onDetailsDisplay.bind(this);
+  }
+
+  render() {
+    const test = this.props.test,
+          cx = classNames,
+          classes = cx({
+            test: true,
+            toggled: this.state.toggled
+          });
+    let paths,
+        details = {};
+
+    if (test.result) {
+      paths = {
+        baseline: test.result.baseline
+      };
+
+      if (test.result.screenshot && test.result.diff) {
+        paths.diff = test.result.diff;
+        paths.screenshot = test.result.screenshot;
+      }
+
+      details = {
+        paths: paths,
+        error: this.props.test.error
+      };
     }
 
     return <div className={classes}>
-      <p className='test-title' onClick={this.displayDetails}>{test.title}
-        <span className='test-state'>{test.state}</span>
+      <p className="test-title" onClick={this.onDetailsDisplay}>
+        {test.state === 'passed'
+            ? <span className="glyphicon glyphicon-ok green"
+                    aria-hidden="true">
```

Comment:
We don't really need aria attributes. They're accessibility related and we have no plans for that now.